_posts/2024-06-14-devlog-28.md

```diff
@@ -0,0 +1,131 @@
+
+As a close friend of Scott's, I've been following Automation Station since its inception as a conveyor belt blockage math problem, feverishly being discussed in the backchannels of Discord by a few heady nerds between binge sessions of Factorio and Rocket League. Who knew what some ASCII would turn into years later? 
+
+{% include gallery id="gallery" layout="half" caption="Conveyor Belt Ideas" %}
+
+
+Automation Station has come a long way since then, and I'm excited to be joining Scott in trying to make the best cozy automation game we possibly can.
+
+With that out of the way, let's dive into a couple new features I've been working on.
+
+## A Living World
+
+One question I think about often when judging the quality of a game is "How does this game make the player feel"? In my experience, if I'm not feeling anything while playing a game, that apathy is going to lead to me setting the game down, likely forever. The best games are real works of art: they evoke *emotion*. They make people throw controllers through their TV screens in a way a painting never could. They fill players with pride over their achievements. They soothe the anxiety of long, stressful days. And the game developer as an artist has to be mindful of what they're creating and what emotions they are evoking. When I've shown Automation Station to people for the first time, they'll see boxbot and the squishy blubber trees and they'll squeal with joy over the aesthetic, even though they know nothing of what the game entails. That visceral, emotional reaction of cute-safe-joy, of oh-my-god-look-at-that-lil-guy, that reaction is exactly what a cozy game should aim for. That engagement with the living world that the player is stepping into is something I wanted to build upon for my first contributions to the game. 
+
+So I made a butterfly:
+
+<video width="100%" autoplay="autoplay" loop="true" muted>
+  <source src="https://i.imgur.com/cNxPs8h.mp4" type="video/mp4" />
+</video>
+
+I figured something small like a butterfly would be a great first step in learning some basics about game dev, like how to make models, materials and apply those to game objects. Dipping my toes in ended up turning into a deep dive on shadergraph, a quick bootcamp on how things are connected in Unity, re-learning some algebra, and watching dozens of butterfly videos on youtube. 
+
+With guidance from Scott, I started on a simple plan:
+1. Make a flat quad model that represented half of a butterfly
+2. Use a shader to cut out a butterfly shape from the quad
+3. Use that same shader to mirror that quad in texture and angle to create the butterfly's other half
+4. Move that quad around to mimic butterfly behavior
```

Comment:
Ackchyually, the butterfly is two quads:
![image](https://github.com/modulocity/automationstationgame.com/assets/2761011/8dd708aa-a98f-45d9-adb1-aff5d420bc28)

The two quad model you made in blender mirrors the UV's on each quad, but each includes the entire UV range which means each wing will have the entire texture. So the texture gets mirrored because of how you UV-mapped the two-quad model, not because of anything fancy in the shader.